cupyx/scipy/sparse/linalg/_eigen.py

```diff
@@ -319,17 +324,41 @@ def svds(a, k=6, *, ncv=None, tol=0, which='LM', maxiter=None,
         raise ValueError('k must be smaller than min(m, n) (actual: {})'
                          ''.format(k))
 
-    aH = a.conj().T
-    if m >= n:
-        aa = aH @ a
+    if isinstance(a, _interface.LinearOperator):
+        if m >= n:
+            a_dot = a.matvec
+            a_matmat = a.matmat
+            aH_dot = a.rmatvec
+            aH_matmat = a.rmatmat
+        else:
+            a_dot = a.rmatvec
+            a_matmat = a.rmatmat
+            aH_dot = a.matvec
+            aH_matmat = a.matmat
     else:
-        aa = a @ aH
+        aH = a.conj().T
+        if m >= n:
+            a_dot = a_matmat = a.dot
+            aH_dot = aH_matmat = aH.dot
+        else:
+            a_dot = a_matmat = aH.dot
+            aH_dot = aH_matmat = a.dot
+
+    def aH_a_matvec(x):
+        return(aH_dot(a_dot(x)))
+
+    def aH_a_matmat(x):
+        return(aH_matmat(a_matmat(x)))
+
+    min_mn = min(m, n)
+    aH_a = _interface.LinearOperator(matvec=aH_a_matvec, matmat=aH_a_matmat,
```

Comment:
This PR changes the behavior for non-`LinearOperator` input because `aH_a` is always of `LinearOperator` type. Is it intended?

Reply:
Yes, this is an intended change, the same policy as `svds` in SciPy.

The advantage of this change is that it reduces the memory usage since there is no need to compute the sparse matrix and sparse matrix multiplication. The disadvantage is that the speedup implemented in `eigsh` when the input matrix is in CSR format cannot be used.

I've adopted this change in this PR for compatibility with SciPy's `svds` implementation, but I will consider how to balance memory usage and performance later.

Reply:
Thanks, I understand the point and agree with your change!
Computing sparse matrix and sparse matrix multiplication may be faster if `n` is so smaller compared to `m` and vice versa, but it's OK to follow the same policy as SciPy's `svds` in this PR.